src/ardupilot/ArduPilotProtocol.h

```diff
@@ -0,0 +1,101 @@
+#include "../network/websocket/WebSocketProtocol.h"
+#include "../network/websocket/WebSocketServer.h"
+#include "../world_interface/data.h"
+
+#include <condition_variable>
+#include <mutex>
+
+namespace net {
+namespace ardupilot {
+
+class ArduPilotProtocol {
+public:
+	ArduPilotProtocol();
+	~ArduPilotProtocol();
+	ArduPilotProtocol(const ArduPilotProtocol& other) = delete;
+	ArduPilotProtocol& operator=(const ArduPilotProtocol& other) = delete;
+
+	// Registers the protocol with the websocket server
+	void initArduPilotServer();
```

Comment:
This looks like UB if this method is called more than once. Is it cleaner to do this in the constructor instead, so we can guarantee that it's only called once?

Reply:
ya this works. its called in the constructor and initArduPilotServer is a private function now